### PR TITLE
deep link differently depending on whether or not the ext is around

### PR DESCRIPTION
**Pull request by camhashemi**
@andrewconner @ryanpbrewster

### CODE REVIEW

File: kifi-backend/modules/shoebox/test/com/keepit/commanders/AuthCommanderTest.scala [hunk 1]
```diff
@@ -122,7 +122,6 @@ class AuthCommanderTest extends Specification with ShoeboxApplicationInjector {
       )
 
       val res1 = authCommander.loginWithTrustedSocialIdentity(identityId)
-      println(res1)
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/test/com/keepit/commanders/AuthCommanderTest.scala:125]: :+1:

**Comment by andrewconner** [comment]
Seems reasonable to me, thanks

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala [hunk 2]
```diff
@@ -107,22 +107,17 @@ class SlackAuthRouter @Inject() (
...
-      case _ => (false, false)
-    }
-
+    val isLoggedIn = request.userIdOpt.isDefined
     // show 3rd party url with ext if possible, otherwise go to keep page (with proper upsells) or 3rd party url
     val redirOpt = db.readOnlyMaster { implicit s =>
       Keep.decodePublicId(pubId)
         .map(keepId => keepRepo.get(keepId)).filter(_.isActive)
         .map { keep =>
           val keepPageUrl = pathCommander.pathForKeep(keep).absolute
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala:116]: Can be placed in `else if` branch below

**Comment by andrewconner** [comment]
:+1:

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala (hunk 2 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala:116]: we use it in two separate `if` branches, the other one right below the diff hehe

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala:116]: Oh I see. Carry on :)

**Comment by camhashemi** [comment]
going to test this baby out

**Pull request merged**